doc/src/guides/solving.md

```diff
@@ -0,0 +1,45 @@
+
+| Description                                                  | Example                                                                                                                     |
+|--------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|
+| [ Solve an equation algebraically ](http://docs.sympy.org)                        | [  $x^2 = 4 $  ](http://docs.sympy.org)                                                                                                       |
+| [ Solve a system of equations algebraically ](http://docs.sympy.org)              | [  $x^2 + y = 2, x - y = 4$  ](http://docs.sympy.org)                                                                                              |
+| [ Solve an equation numerically ](http://docs.sympy.org)                          | [ $cos(x) = x $ ](http://docs.sympy.org)                                                                                                     |
+| [ Solve a partial differential equation algebraically ](http://docs.sympy.org)    | [${\frac {\partial ^{2}(x^3 y^3)}{\partial x\partial y}}=9 $](http://docs.sympy.org)                                                                     |
+| [ Solve an ordinary differential equation algebraically  ](http://docs.sympy.org) | [ $f\prime\prime(x) + 9f(x)=0 $ ](http://docs.sympy.org)                                                    |
+| [ Solve a system of linear equations algebraically ](http://docs.sympy.org)       | [  $x + y = 2, x - y = 0$  ](http://docs.sympy.org)                                                                                           |
+| [ Solve a system of nonlinear equations algebraically ](http://docs.sympy.org)    | [  $x^2 + y^3 = 1, x^3 - y^2 = 0 $  ](http://docs.sympy.org)                                                                                      |
+| [ Solve a matrix problem algebraically ](http://docs.sympy.org)                   | [ $ \left[\begin{array}{cc} 1 & 1\\1 & -1\end{array}\right] \left[\begin{array}{cc} x\\y\end{array}\right] = \left[\begin{array}{cc} 2\\0\end{array}\right] $ ](http://docs.sympy.org) |
+| [ Solve an inequality algebraically ](http://docs.sympy.org)                      | [ $ x^2 < 4 $ ](http://docs.sympy.org)                                                                                                        |
+| [ Solve a system of inequalities algebraically ](http://docs.sympy.org)           | [ $ x^2 < 4, x > 0 $ ](http://docs.sympy.org)                                                                                                 |
+| [ Solve a polynomial algebraically ](http://docs.sympy.org)                       | [ $ x^2 - x = 0 $ ](http://docs.sympy.org)                                                                                                |
```

Comment:
I would use the word roots here: "Find roots of a polynomial".

Reply:
I think it would also be good to have a page that discusses solving systems of polynomials with Groebner bases.

Reply:
The only search query in the top 1000 containing _groebner_ is

Top queries | Clicks | Impressions | CTR | Position
-- | -- | -- | -- | --
sympy _groebner_ basis | 60 | 76 | 78.95% | 1

So while Groebner bases are conceptually important, I'm not sure there's enough Google searches (which is a proxy for overall interest because we don't collect site analytics) for them to justify including them on this page, or at least in the tranche of types planned for the Season of Docs project. A page discussing the technique may well be useful, though again is probably beyond the scope of this project.

Reply:
Here are the [search queries in the top 1000](https://docs.google.com/spreadsheets/d/1zuQDU1Mj8Ua8zAAxRu5hr7b23wCRlIJ2cOK6FPMnJq0/edit?usp=sharing) involving _polynomial_, with queries involving **solve** or **roots** bolded:

Top queries | Clicks | Impressions | CTR | Position
-- | -- | -- | -- | --
polynomial manipulation | 240 | 1763 | 13.61% | 2.88
sympy polynomial | 1079 | 1574 | 68.55% | 1
sympy polynomial coefficients | 155 | 406 | 38.18% | 1.35
sympy polynomial division | 254 | 365 | 69.59% | 1
**sympy roots of polynomial** | 238 | **365** | 65.21% | 1.04
polynomial manipulation in data structure | 74 | 344 | 21.51% | 4.34
sympy characteristic polynomial | 213 | 336 | 63.39% | 1
sympy factor polynomial | 157 | 330 | 47.58% | 1.03
**sympy solve polynomial** | 218 | **300** | 72.67% | 1
sympy polynomials | 171 | 242 | 70.66% | 1
sympy factorize polynomial | 162 | 219 | 73.97% | 1
sympy expand polynomial | 126 | 211 | 59.72% | 1
sympy coefficients of polynomial | 93 | 175 | 53.14% | 1.18
**sympy polynomial roots** | 90 | **144** | 62.50% | 1.28
sympy polynomial from coefficients | 52 | 138 | 37.68% | 1.26

So there are more searches involving **roots** than **solve**, 509 vs. 365. Given those two terms are used with fairly comparable frequency, I think a description including both is worthwhile from the perspective of both search engine optimization and user recognition:

`Solve (find the roots of) a polynomial algebraically`

Reply:
"Groebner" is somewhat analogous to the Diophantine/integer equations situation. It's the technical name for the algorithm used to solve systems of polynomial equations. It might be helpful to include "Groebner" in the title text, but the main thing people will look for is systems of equations.

Also, in my experience, people won't necessarily recognize that they have a "system of polynomial equations" vs. just a "system of equations". It partially depends on how mathematically sophisticated they are. But also SymPy itself makes the distinction because the methods used to solve them are different, but people who aren't familiar with those methods might not make the same distinctions. For example, Groebner bases make it straightforward to solve a system of *polynomial* equations, but if the system includes a non-polynomial equation it is much harder because the algorithm no longer applies. But people may not appreciate that and might not even realize that their system is in fact non-polynomial. These are the sorts of caveats we will have to include in the solvers guides themselves.

Reply:
I often see polynomial equations being described as "nonlinear". It might be good to clarify when talking about nonlinear equations that polynomials are an important special case.